nigsp/operations/statistics.py

```diff
@@ -0,0 +1,195 @@
+import logging
+
+import numpy as np
+from mne.stats import permutation_t_test
+
+LGR = logging.getLogger(__name__)
+
+
+def ranktest(a, axis=None):
+    # Code adapted from `scipy`; ref: https://github.com/scipy/scipy/blob/v1.11.2/scipy/stats/_stats_py.py#L10123
+
+    """Assign ranks to data, dealing with ties appropriately.
+
+    By default (``axis=None``), the data array is first flattened, and a flat
+    array of ranks is returned. Separately reshape the rank array to the
+    shape of the data array if desired (see Examples).
+
+    Ranks begin at 1.  The `method` argument controls how ranks are assigned
+    to equal values.
+
+    Parameters
+    ----------
+    a : array_like
+        The array of values to be ranked.
+    axis : {None, int}, optional
+        Axis along which to perform the ranking. If ``None``, the data array
+        is first flattened.
+
+    Returns
+    -------
+    ranks : ndarray
+         An array of size equal to the size of `a`, containing rank
+         scores.
+    """
+
+    if axis is not None:
+        a = np.asarray(a)
+        return np.apply_along_axis(ranktest, axis, a)
+
+    arr = np.ravel(np.asarray(a))
+    algo = "quicksort"
+    sorter = np.argsort(arr, kind=algo)
+    inv = np.empty(sorter.size, dtype=np.intp)
+    inv[sorter] = np.arange(sorter.size, dtype=np.intp)
+
+    arr = arr[sorter]
+    obs = np.r_[True, arr[1:] != arr[:-1]]
+    dense = obs.cumsum()[inv]
+
+    # cumulative counts of each unique value
+    count = np.r_[np.nonzero(obs)[0], len(obs)]
+
+    # average method
+    return 0.5 * (count[dense] + count[dense - 1] + 1)
+
+
+# Code adapted from `mne-python`; ref: https://mne.tools/stable/generated/mne.stats.ttest_1samp_no_p.html
+def ttest_1samp_no_p(X, axis=0):
+    """Perform one-sample t-test.
+
+    This is a modified version of :func:`scipy.stats.ttest_1samp` that avoids
```

Comment:
And because it's then a derivative of a derivative, we need to report all licences all around (both scipy and MNE)